test/bases/renku_data_services/data_api/test_sessions.py

```diff
@@ -192,6 +205,57 @@ async def test_patch_session_environment(
     assert res.json.get("mount_directory") is None
 
 
+@pytest.mark.asyncio
+async def test_patch_session_environment_archived(
+    sanic_client: SanicASGITestClient,
+    admin_headers,
+    create_session_environment,
+    create_project,
+    valid_resource_pool_payload,
+    create_resource_pool,
+) -> None:
+    env = await create_session_environment("Environment 1")
+    environment_id = env["id"]
+
+    payload = {"is_archived": True}
+
+    _, res = await sanic_client.patch(f"/api/data/environments/{environment_id}", headers=admin_headers, json=payload)
+
+    assert res.status_code == 200, res.text
+    assert res.json is not None
+    assert res.json.get("is_archived")
+
+    # Test that you can't create a launcher with an archived environment
```

Comment:
Question: how does this interact with copying a project?

Reply:
I'm still testing this a bit in the deployment to see how it interacts with other things and make sure it doesn't break anything